module/helpers/attachmentHelper.mjs

```diff
@@ -0,0 +1,196 @@
+        const attachmentSource = effect.flags?.daggerheart?.attachmentSource;
+        return attachmentSource && 
+               attachmentSource[parentUuidProperty] === parentItem.uuid && 
+               attachmentSource.itemUuid === attachedUuid;
+    });
+
+    if (effectsToRemove.length > 0) {
+        await actor.deleteEmbeddedDocuments('ActiveEffect', effectsToRemove.map(e => e.id));
+    }
+}
+
+/**
+ * Remove an attachment from an item and clean up its effects
+ * @param {Object} options - Configuration options
+ * @param {Item} options.parentItem - The item (armor/weapon) that the item is attached to
+ * @param {string} options.attachedUuid - UUID of the attached item being removed
+ * @param {string} options.parentType - Type of parent item ("armor" or "weapon")
+ * @returns {Promise<void>}
+ */
+export async function removeAttachmentFromItem({ parentItem, attachedUuid, parentType }) {
+    const currentAttached = parentItem.system.attached;
+    
+    // Remove the attachment from the parent item's attached array
+    await parentItem.update({
+        'system.attached': currentAttached.filter(uuid => uuid !== attachedUuid)
+    });
+    
+    // Remove any effects that came from this attachment
+    await removeAttachmentEffectsFromActor({
+        parentItem,
+        attachedUuid,
+        parentType
+    });
+}
+
+/**
+ * Handle adding/removing attachment effects when a parent item is equipped/unequipped
+ * @param {Object} options - Configuration options
+ * @param {Item} options.parentItem - The item (armor/weapon) being equipped/unequipped
+ * @param {boolean} options.newEquippedStatus - The new equipped status
+ * @param {string} options.parentType - Type of parent item ("armor" or "weapon")
+ * @returns {Promise<void>}
+ */
+export async function handleAttachmentEffectsOnEquipChange({ parentItem, newEquippedStatus, parentType }) {
+    // Try to get the actor - it might be parentItem.parent instead of parentItem.parent.parent
+    const actor = parentItem.parent?.type === 'character' ? parentItem.parent : parentItem.parent?.parent;
+    
+    if (!actor || !parentItem.system.attached?.length) {
+        return;
+    }
+
+    if (newEquippedStatus) {
+        // Item is being equipped - add attachment effects
+        for (const attachedUuid of parentItem.system.attached) {
+            const attachedItem = await fromUuid(attachedUuid);
+            if (attachedItem && attachedItem.effects.size > 0) {
+                await copyAttachmentEffectsToActor({
+                    parentItem,
+                    attachedItem,
+                    attachedUuid,
+                    parentType
+                });
+            }
+        }
+    } else {
+        // Item is being unequipped - remove attachment effects
+        const parentUuidProperty = `${parentType}Uuid`;
+        const effectsToRemove = actor.effects.filter(effect => {
+            const attachmentSource = effect.flags?.daggerheart?.attachmentSource;
```

Comment:
Flags can be defined in `flagsConfig.mjs`.
More stable way of getting the flag without magic strings in different places in the code.
Something like: `effect.getFlag(CONFIG.DH.id, CONFIG.DH.FLAGS.itemAttachmentSource);`